### PR TITLE
Return time series data by source [backend only]

### PR DESCRIPTION
**Pull request by avalmas-programize**
Group time series data by source.
Returning object:
```js
{
  metric: {
    hobo: [data]
    spotter: [data]
  }
}
```

### CODE REVIEW

**Comment by ericboucher** [comment]
@tleraj can you make the necessary code changes in the frontend?

File: packages/api/src/time-series/time-series.service.ts [hunk 1]
```diff
@@ -45,36 +58,60 @@ export class TimeSeriesService {
...
+          .andWhere('time_series.reef_id = :reefId', { reefId })
           .andWhere(poiCondition)
           .andWhere('timestamp >= :startDate', { startDate })
           .andWhere('timestamp <= :endDate', { endDate })
           .orderBy('timestamp', 'ASC')
           .getRawMany();
   }
 
+  private getDataRangeQuery(reefId: number, poiId?: number) {
+    const poiCondition = poiId
```
Comment (ericboucher) [inline packages/api/src/time-series/time-series.service.ts:98]: Let's also keep it like this

`const poiCondition = poiId ? `poi_id = ${poiId}` : 'poi_id is NULL';` no?

File: packages/api/src/time-series/time-series.service.ts [hunk 2]
```diff
@@ -45,36 +58,60 @@ export class TimeSeriesService {
     reefId: number,
     poiId?: number,
   ): Promise<TimeSeriesData[]> {
-    const poiCondition = poiId ? `poi_id = ${poiId}` : 'poi_id is NULL';
```
[the remark this answers is not in the document]
Comment (avalmas-programize) [reply packages/api/src/time-series/time-series.service.ts:48]: I changed that so on when specifying a poi on your params you will also get the data of the reef (poi_id is NULL)

File: packages/api/src/time-series/time-series.service.ts (hunk 1 above)
Comment (avalmas-programize) [reply packages/api/src/time-series/time-series.service.ts:98]: Same as above. By grabbing both poiId values and NULL we are getting the ranges for the poi and also for the reef

File: packages/api/src/time-series/time-series.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ericboucher) [reply packages/api/src/time-series/time-series.service.ts:48]: ok but then it's grouped by source, but the source for the reef is called "null"?

**Review by ericboucher — APPROVED** [review]

File: packages/api/src/time-series/time-series.service.ts (hunk 1 above)
Comment (ericboucher) [reply packages/api/src/time-series/time-series.service.ts:99]: Follow #495 for a future resolution

**Pull request merged**